pkg/daemon/controller/controller.go

```diff
@@ -370,11 +393,18 @@ func (sc *ServiceController) resolveNodeVariables(svc servicescm.Service) (map[s
 	return vars, nil
 }
 
-// resolvePowershellVariables returns a map, with the keys being each variable, and the value being the string to replace the
-// variable with
+// resolvePowershellVariables returns a map, with the keys being each variable, and the value being the string to
+// replace the variable with
 func (sc *ServiceController) resolvePowershellVariables(svc servicescm.Service) (map[string]string, error) {
-	// TODO: Implement this function
-	return make(map[string]string), nil
+	vars := make(map[string]string)
+	for _, psVar := range svc.PowershellVariablesInCommand {
+		out, err := sc.psCmdRunner.Run(psVar.Path)
```

Comment:
For executing PowerShell scripts from a file `powershell.exe -File` may be better equipped. See PowerShell [docs](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_powershell_exe?view=powershell-5.1#-file----filepath-args).

With that, I feel you may need two APIs for `psCmdRunner`:
- psCmdRunner.Run
- psCmdRunner.RunFile